### PR TITLE
Added create category page

### PR DESCRIPTION
**Pull request by Icon1cc**

### CODE REVIEW

File: src/services/categories.ts [hunk 1]
```diff
@@ -18,9 +18,16 @@ const list = (
     query["order_by"] = sortField;
   }
 
-  return get("categories", query);
+  return get("/categories", query);
```
Comment (DoStini) [inline src/services/categories.ts:21]: Why the initial `/`?  Wasnt it working before?

File: src/services/categories.ts [hunk 2]
```diff
@@ -18,9 +18,16 @@ const list = (
...
+};
+
+const create = (name: string) => {
+  const body = { name };
+  return jsonPost("/categories", body);
```
Comment (DoStini) [inline src/services/categories.ts:26]: is the initial `/` necessary?  I dont think so

File: src/services/categories.ts [hunk 3]
```diff
@@ -18,9 +18,16 @@ const list = (
...
 };
 
 export default {
   list,
+  create,
+  // ... any other service functions ...
```
Comment (DoStini) [inline src/services/categories.ts:32]: remove

File: src/components/tables/CategoriesList.tsx [hunk 4]
```diff
@@ -96,11 +99,11 @@ export default function CategoriesList() {
...
-          className="bg-custom-dark-blue flex text-white px-3 py-2 rounded font-medium items-center justify-center text-sm hover:bg-custom-hover-blue transition"
-          onClick={() => alert("Create new category")}
-        >
-          <FaPlus /> <span className="ml-2">New category </span>
-        </button>
+           className="bg-custom-dark-blue flex text-white px-3 py-2 rounded font-medium items-center justify-center text-sm hover:bg-custom-hover-blue transition"
+            onClick={() => navigate("/category/create")}
+            >
+             <FaPlus /> <span className="ml-2">New category </span>
+          </button>
```
Comment (DoStini) [inline src/components/tables/CategoriesList.tsx:106]: If the onClick is a navigate, please use a `Link` component

File: src/components/tables/CategoriesList.tsx [hunk 5]
```diff
@@ -47,6 +48,8 @@ export default function CategoriesList() {
   } | null>(null);
   const [filter, setFilter] = useState("");
 
+  const navigate = useNavigate();
```
Comment (DoStini) [inline src/components/tables/CategoriesList.tsx:51]: If the only call is in the button, remove

File: src/App.tsx [hunk 6]
```diff
@@ -43,6 +44,12 @@ function App() {
             path="/exercises/create"
             element={<ProtectedRoute element={<CreateExercisePage />} />}
           />
+
+            <Route
+            path="/category/create"
```
Comment (DoStini) [inline src/App.tsx:49]: Let's be uniform (everything is in plurar)

```suggestion
            path="/categories/create"
```

File: src/components/CreateCategory.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    categoriesService
+      .create(name)
+      .then(() => {
+        navigate("/categories");
+      })
+      .catch((err) => {
+        console.log(err);
```
Comment (DoStini) [inline src/components/CreateCategory.tsx:19]: Should add the error component aswell (after we merge it)

Comment (DoStini) [reply src/components/CreateCategory.tsx:19]: Verify which responses this route might give as an error

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/components/CreateCategory.tsx (hunk 7 above)
Comment (DoStini) [reply src/components/CreateCategory.tsx:19]: Add the snackbar

**Review by DoStini — APPROVED** [review]

File: src/services/categories.ts [hunk 8]
```diff
@@ -21,6 +21,12 @@ const list = (
   return get("categories", query);
 };
 
+const create = (name: string) => {
+  const body = { name };
+  return jsonPost("categories", body);
+};
```
Comment (DoStini) [inline src/services/categories.ts:27]: You should've tested things locally, this is not working  because you are calling the endpoint wrong

```suggestion
const create = (name: string) => {
  return jsonPost(`categories/${name}`, body);
};
```

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 9]
```diff
@@ -70,9 +71,14 @@ function App() {
             }
           />
 
-          <Route path="/404" element={<h1>404</h1>} />
```
Comment (DoStini) [inline src/App.tsx:73]: Restore this routes

File: src/App.tsx [hunk 10]
```diff
@@ -70,9 +71,14 @@ function App() {
...
+            <Route
+            path="/categories/create"
+            element={<ProtectedRoute element={<CreateCategory />} 
+            permissionLevel={ADMIN_PERMISSIONS.ADMIN}
+            />
+          }
+          />
 
-          <Route path="*" element={<Navigate to="/404" />} />
```
Comment (DoStini) [inline src/App.tsx:75]: Restore this routes

File: src/components/CreateCategory.tsx [hunk 11]
```diff
@@ -0,0 +1,78 @@
+// CreateCategory.tsx
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import categoriesService from "../services/categories";
+import Alert from "./Alert"; 
+
+const CreateCategory: React.FC = () => {
```
Comment (DoStini) [inline src/components/CreateCategory.tsx:7]:
```suggestion
const CreateCategory = () => {
```

File: src/components/CreateCategory.tsx [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  return (
+    <div className="bg-white min-h-screen flex flex-col">
+      <div className="w-full py-4 px-6 bg-custom-light-grey border-b border-gray-200">
+        <h1 className="text-custom-dark-blue text-xl font-bold">
+          Create Category
+        </h1>
+      </div>
+      <div className="flex-grow flex items-center justify-center p-8">
+        <div className="bg-custom-light-grey p-12 rounded-lg shadow-xl w-full max-w-2xl">
+          {/* Conditionally render the Alert component based on showAlert state */}
```
Comment (DoStini) [inline src/components/CreateCategory.tsx:36]:
```suggestion
```

File: src/components/CreateCategory.tsx [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+                type="submit"
+                className="px-6 py-3 bg-custom-dark-blue text-white text-lg rounded hover:bg-blue-800"
+              >
+                Save
+              </button>
+              <button
+                type="button"
+                onClick={() => navigate("/categories")}
+                className="px-6 py-3 bg-custom-grey text-gray-800 text-lg rounded hover:bg-gray-400"
+              >
```
Comment (DoStini) [inline src/components/CreateCategory.tsx:67]: if a button is inClick navigate, please use Link to categories

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/components/CreateCategory.tsx [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  const [name, setName] = useState("");
+  const [alertMessage, setAlertMessage] = useState(""); 
+  const [showAlert, setShowAlert] = useState(false); 
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    categoriesService
+      .create(name)
+      .then(() => {
+      })
```
Comment (DoStini) [inline src/components/CreateCategory.tsx:17]: Why did you remove the redirect after success?

File: src/services/categories.ts [hunk 15]
```diff
@@ -21,6 +21,11 @@ const list = (
   return get("categories", query);
 };
 
+const create = (name: string) => {
+  return jsonPost(`categories/${name}`, { name });
```
Comment (DoStini) [inline src/services/categories.ts:25]:
```suggestion
  return jsonPost(`categories/${name}`);
```

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 9 above)
Comment (Icon1cc) [reply src/App.tsx:73]: Restored

File: src/components/CreateCategory.tsx (hunk 14 above)
Comment (Icon1cc) [reply src/components/CreateCategory.tsx]: Added it back

File: src/services/categories.ts (hunk 15 above)
Comment (Icon1cc) [reply src/services/categories.ts]: Fixed it

File: src/components/CreateCategory.tsx (hunk 13 above)
Comment (Icon1cc) [reply src/components/CreateCategory.tsx]: Changed to link

File: src/App.tsx (hunk 10 above)
Comment (Icon1cc) [reply src/App.tsx:75]: Restored

File: src/components/CreateCategory.tsx (hunk 11 above)
Comment (Icon1cc) [reply src/components/CreateCategory.tsx]: Fixed it

File: src/components/CreateCategory.tsx (hunk 12 above)
Comment (Icon1cc) [reply src/components/CreateCategory.tsx]: removed

**Review by DoStini — APPROVED** [review]
